integrations/google_vertex/tests/test_gemini.py

```diff
@@ -0,0 +1,212 @@
+                                    },
+                                    "unit": {"type_": "STRING", "enum": ["celsius", "fahrenheit"]},
+                                },
+                                "required": ["location"],
+                            },
+                        }
+                    ]
+                }
+            ],
+        },
+    }
+
+
+@patch("haystack_integrations.components.generators.google_vertex.gemini.vertexai")
+@patch("haystack_integrations.components.generators.google_vertex.gemini.GenerativeModel")
+def test_from_dict(_mock_vertexai, _mock_generative_model):
+    gemini = VertexAIGeminiGenerator.from_dict(
+        {
+            "type": "haystack_integrations.components.generators.google_vertex.gemini.VertexAIGeminiGenerator",
+            "init_parameters": {
+                "project_id": "TestID123",
+                "model": "gemini-pro-vision",
+                "generation_config": {
+                    "temperature": 0.5,
+                    "top_p": 0.5,
+                    "top_k": 0.5,
+                    "candidate_count": 1,
+                    "max_output_tokens": 10,
+                    "stop_sequences": ["stop"],
+                },
+                "safety_settings": {HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_ONLY_HIGH},
+                "tools": [
+                    {
+                        "function_declarations": [
+                            {
+                                "name": "get_current_weather",
+                                "description": "Get the current weather in a given location",
+                                "parameters": {
+                                    "type_": "OBJECT",
+                                    "properties": {
+                                        "location": {
+                                            "type_": "STRING",
+                                            "description": "The city and state, e.g. San Francisco, CA",
+                                        },
+                                        "unit": {"type_": "STRING", "enum": ["celsius", "fahrenheit"]},
+                                    },
+                                    "required": ["location"],
+                                },
+                            }
+                        ]
+                    }
+                ],
+                "streaming_callback": None,
+            },
+        }
+    )
+
+    assert gemini._model_name == "gemini-pro-vision"
+    assert gemini._project_id == "TestID123"
+    assert gemini._safety_settings == {HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_ONLY_HIGH}
+
+    # assert gemini._tools == [Tool(function_declarations=[GET_CURRENT_WEATHER_FUNC])]
+    assert isinstance(gemini._generation_config, GenerationConfig)
+
+
+@patch("haystack_integrations.components.generators.google_vertex.gemini.GenerativeModel")
```

Comment:
Why are we not mocking `@patch("haystack_integrations.components.generators.google_vertex.gemini.vertexai.init")` here and in `test_run_with_streaming_callback` too?